internal/datatransfer/monitor.go

```diff
@@ -49,6 +58,8 @@ func (m *Monitor) GetLastSuccessfulSyncTime(workloadName string) *time.Time {
 }
 
 func (m *Monitor) WorkloadRemoved(workloadName string) {
+	m.syncPaths()
```

Comment:
It should be triggered for one specific workload, not all of them.

Reply:
Im not sure it's better that 
If we will add a map in the configuration.Manager we will need to update that map in each create/remove/update workload - it will be place that holds the data on workloads (this will be a duplicate struct)

Reply:
According to the sequence of actions of wrapper.Remove(workload) - something will now work properly here.
I'll try to simplify it by removing unnecessary code:
```
func (ww workloadWrapper) Remove(workloadName string) error {
	ww.workloads.Remove(workload)
	ww.mappingRepository.Remove(workloadName)

	for _, observer := range ww.observers {
		observer.WorkloadRemoved(workloadName)
	}
}
```

meaning, the call for `m.syncPaths()` is being invoked only after the workload is removed.
however, `m.syncPaths()` starts with:
```
func (m *Monitor) syncPaths() {
	workloads, err := m.workloads.ListWorkloads()
	if len(workloads) == 0 {
		return
	}
```
At least for the last workload, this will quit without last sync for its data.

I'd try to simplify it and don't attempt to sync data as part of each remove.
After all, we'd might end up running `syncPaths()` here in parallel to the one being invoked by the ticker.

I can suggest an alternative approach so we won't have to mess to much with this code and make it too complex. See my other comment on manager.Deregister: https://github.com/jakub-dzon/k4e-device-worker/pull/16/files#diff-ab1f64e13b34d19039deb0762b992d04e2ece198e03ee84091e3df02a8d776b3R101